src/cbuild/hooks/post_install/010_completions.py

```diff
@@ -0,0 +1,53 @@
+def invoke(pkg):
+    comps = []
+
+    if (
+        pkg.pkgname != "fish-shell"
+        and (compspath := pkg.destdir / "usr/share/fish/completions").is_dir()
+    ):
+        comps.extend(
+            (f, f.name.removesuffix(".fish")) for f in compspath.iterdir()
+        )
+
+    if (
+        pkg.pkgname != "fish-shell"
+        and (
+            compspath := pkg.destdir / "usr/share/fish/vendor_completions.d"
+        ).is_dir()
+    ):
```

Comment:
fish-shell only ships the top one, so the first part of this check doesn't do anything (when it's fish-shell, this is also empty)

for the top one, we have actual packages linted out from putting stuff in the top path too, don't we?

so i think this should just be

```
if pkg.pkgname in ["fish-shell", "bash-completion", "zsh"]:
  return
```

or, better- add an options= !lintcompletions and use that instead

Reply:
> fish-shell only ships the top one, so the first part of this check doesn't do anything (when it's fish-shell, this is also empty)

good point yes, missed that

> for the top one, we have actual packages linted out from putting stuff in the top path too, don't we?

i couldn't find any lint for that, where is it?

Reply:
nvm we don't

```
def _split_fishcomp(pkg):
    pkg.take("usr/share/fish/completions", missing_ok=True)
    pkg.take("usr/share/fish/vendor_completions.d", missing_ok=True)
```

seems it just takes it. i guess that's fine :)  

in that case forget what i said for that

Reply:
actually, according to https://fishshell.com/docs/current/completions.html#where-to-put-completions:

> The completions shipped with fish, usually installed in /usr/share/fish/completions

so custom completions should go in the vendor completions dir, and we don't need to consider `usr/share/fish/completions` here either

Reply:
i already know the docs, but my previous experience was random apps put it in fish/completions anyway.

we could lint that too and do stuff but in general i don't think that will have much benefit. at best, it avoids a later file conflict when fish itself ships a completion for the same name, and it happens to clash. if not for that though, it makes zero runtime difference and just involves moving files around, and hopefully upstreaming it.

but from a quick check nothing breaks this currently, so i guess it's ok, feel free to add that in a separate mr

Reply:
opened #2768 for that